File: sdk/node/src/types/audit.ts
export interface AuditLogQuery {
  tenant_id?: string;
  event_types?: string[];
  event_categories?: string[];
  actor_id?: string;
  target_id?: string;
  status?: string;
  from_date?: string;
  to_date?: string;
  limit?: number;
  offset?: number;
}

export interface AuditLog {
  id: string;
  tenant_id: string;
  event_type: string;
  event_category?: string;
  event_action?: string;
  actor_type?: string;
  actor_id?: string;
  actor_name?: string;
  actor_ip_address?: string;
  target_type?: string;
  target_id?: string;
  target_name?: string;
  description?: string;
  metadata?: Record<string, any>;
  status?: string;
  error_message?: string;
  created_at?: string;
}

export interface AuditLogsResponse {
  logs: AuditLog[];
  total: number;
  limit: number;
  offset: number;
}

export interface AuditStats {
  [key: string]: any;
}

File: sdk/python/coreauth/types/audit.py
from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel


class AuditLogQuery(BaseModel):
    tenant_id: Optional[str] = None
    event_types: Optional[list[str]] = None
    event_categories: Optional[list[str]] = None
    actor_id: Optional[str] = None
    target_id: Optional[str] = None
    status: Optional[str] = None
    from_date: Optional[str] = None
    to_date: Optional[str] = None
    limit: Optional[int] = None
    offset: Optional[int] = None


class AuditLog(BaseModel):
    id: Optional[str] = None
    tenant_id: Optional[str] = None
    event_type: Optional[str] = None
    event_category: Optional[str] = None
    event_action: Optional[str] = None
    actor_type: Optional[str] = None
    actor_id: Optional[str] = None
    actor_name: Optional[str] = None
    actor_ip_address: Optional[str] = None
    actor_user_agent: Optional[str] = None
    target_type: Optional[str] = None
    target_id: Optional[str] = None
    target_name: Optional[str] = None
    description: Optional[str] = None
    metadata: Optional[dict[str, Any]] = None
    status: Optional[str] = None
    error_message: Optional[str] = None
    request_id: Optional[str] = None
    session_id: Optional[str] = None
    created_at: Optional[str] = None


class AuditLogsResponse(BaseModel):
    logs: list[AuditLog] = []
    total: int
    limit: int
    offset: int


class AuditStats(BaseModel):
    data: Optional[dict[str, Any]] = None
